### PR TITLE
Zlib: Update to 1.2.12

### PR DESCRIPTION
**Pull request by eschnett**
This is a security update.

### CODE REVIEW

File: Z/Zlib/build_tarballs.jl [hunk 1]
```diff
@@ -41,6 +41,6 @@ dependencies = Dependency[
 
 # Note: we explicitly lie about this because we don't have the new
 # versioning APIs worked out in BB yet.
-version = v"1.2.12"
+version = v"1.2.13"
```
[the remark this answers is not in the document]
Comment (eschnett) [reply Z/Zlib/build_tarballs.jl:44]: How would Julia installs know that `Zlib_jll` should be updated? Are you saying that they won't be updated unless a new Julia version is installed?

[the remark this answers is not in the document]
Comment (eschnett) [reply Z/Zlib/build_tarballs.jl:44]: I opened this PR on the web interface. Modifying PRs is then tedious. I opened a new one.

[the remark this answers is not in the document]
Comment (eschnett) [reply Z/Zlib/build_tarballs.jl:44]: Are you going to open the respective `JuliaLang/julia` PR?

[the remark this answers is not in the document]
Comment (giordano) [reply Z/Zlib/build_tarballs.jl:44]: I can do that later tonight if you want, just let me know :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (eschnett) [reply Z/Zlib/build_tarballs.jl:44]: Please do. I will watch and learn.

[the remark this answers is not in the document]
Comment (giordano) [reply Z/Zlib/build_tarballs.jl:44]: Sorry, I got derailed by other projects and couldn't do the PR today. I'll try to remember to do it tomorrow, but please feel free to ping me in ~20 hours if I forget

**Pull request closed without merging**